test/eachobs.jl

```diff
@@ -11,7 +11,7 @@
     for (i,x) in enumerate(eachobs(X, buffer=b))
         @test x == X[:,i]
     end
-    @test b == X[:,end]
+    @test_broken b == X[:,end]
```

Comment:
this is the only test that is now failing with this PR. Why is that? It seems we don't use the buffer at all...

Reply:
@lorenzoh what changed here?

Reply:
See https://github.com/JuliaML/MLUtils.jl/pull/82#issuecomment-1113995362; I wasn't sure what the right semantics here are, so I didn't implement the preallocated buffer handling. See also https://github.com/JuliaML/MLUtils.jl/pull/82#issuecomment-1114000506.

Reply:
Is this change okay?

I wouldn't consider this breaking in any case, since `ret = getobs!(buf, data, i)` will fall back to `getobs`, so one should always be using the return value and not rely on `buf` being mutated.